hbase-client/src/test/java/org/apache/hadoop/hbase/client/trace/hamcrest/SpanDataMatchers.java

```diff
@@ -46,6 +48,18 @@ private SpanDataMatchers() { }
     };
   }
 
+  public static <T extends Comparable<T> & TemporalAmount> Matcher<SpanData> hasDuration(
+    Matcher<T> matcher
+  ) {
+    return new FeatureMatcher<SpanData, T>(matcher, "SpanData having duration that ", "duration") {
+      @Override
+      @SuppressWarnings("unchecked")
+      protected T featureValueOf(SpanData item) {
```

Comment:
Looking at the implementation of java.time.Duration, it seems that it can only be compared with another Duration, so do we need to introduce a generic type here? Just use Duration directly?

Reply:
Sure, good point. I guess the `T extends Comparable<T> & TemporalAmount` stuff is overly broad.